tests/e2e/restart_test.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This test suite includes test cases involving restart docker and plugin. #1252
+// End goal is to make sure volumes get detached smoothly in different scenarios
+
+package e2e
+
+import (
+	"os"
+	"testing"
+
+	. "gopkg.in/check.v1"
+
+	dockercli "github.com/vmware/docker-volume-vsphere/tests/utils/dockercli"
+	testparams "github.com/vmware/docker-volume-vsphere/tests/utils/inputparams"
+	vf "github.com/vmware/docker-volume-vsphere/tests/utils/verification"
+)
+
+// Hook up gocheck into the "go test" runner.
+func TestRestart(t *testing.T) { TestingT(t) }
+
+type PluginSuite struct {
+	volumeName    string
+	hostName      string
+	containerName string
+}
+
+func (s *PluginSuite) SetUpTest(c *C) {
+	s.volumeName = testparams.GetVolumeNameWithTimeStamp("abc")
+	s.hostName = os.Getenv("VM2")
```

Comment:
Was debugging failure on CI, if it is photon(vm1) or ubuntu(vm2) issue.
It wasn't the reason. This can be either VM